script.js

```diff
@@ -0,0 +1,149 @@
+}
+
+function getSelectedCategory() {
+    return [...headerNavbar.querySelectorAll('input')].find(item => item.checked).getAttribute('id');
+}
+
+pages.addEventListener('click', ({target}) => {
+    const pageItem = target.closest('.page-item');
+    if (!pageItem.classList.contains('.active')) {
+        pages.querySelectorAll('li').forEach(item => item.classList.remove('active'));
+        pageItem.classList.add('active');
+        const selectedCategory = getSelectedCategory();
+        getData(`${selectedCategory}/?page=${pageItem.getAttribute('data-index')}`)
+            .then((data) => renderCards(data, selectedCategory));
+    }
+})
+
+
+resultBlock.addEventListener('click', (event) => {
+    if (event.target.closest('.btn') && event.isTrusted) {
+        resultBlock.querySelectorAll('.btn').forEach(item => {
+            if (!item.classList.contains('collapsed')) {
+                item.click();
+            }
+        })
+    }
+});
+
+searchButton.addEventListener('click', (event) => {
+    if (!searchMode) {
+        currentPage = [...pages.querySelectorAll('.page-item')].find(item => item.classList.contains('active')).getAttribute('data-index');
+    }
+    if (searchInput.value) {
+        searchMode = true;
+        pages.innerHTML = null;
+        const selectedCategory = getSelectedCategory();
+        getData(`${selectedCategory}/?search=${searchInput.value}`).then((data) => {
+            if (data.count) {
+                renderCards(data, selectedCategory);
+                addPagination(data.count);
+            } else {
+                loader.classList.add(MODIFICATORS.hide);
+                resultBlock.innerHTML += '<span class="h3">Nothing Found</span>';
+            }
+        })
+    }
+})
+
+searchInput.addEventListener('input', () => {
+    if (searchMode && !searchInput.value) {
+        searchMode = false;
+        pages.innerHTML=null;
+        const selectedCategory = getSelectedCategory();
+        getData(selectedCategory).then((data => {
+            renderCards(data, selectedCategory)
+            addPagination(data.count);
+            pages.querySelectorAll('.page-item').forEach(item => {
+                if (item.getAttribute('data-index') === currentPage) {
+                    item.click();
+                }
+            })
+        }))
+    }
+})
+
+function addCardLayout(title, layout, index) {
+    const card = document.createElement('div');
+    card.className = 'card m-2';
+    card.innerHTML = `<div class="card-body">
+            <p class="title">
+                <span class="h5">${title}</span>
+                <button class="btn collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#cardInfo${index}"
+                        aria-expanded="false">
+                    <i class="fas fa-angle-down"></i>
+                </button>
+            </p>
+            <div class="text">
+                ${layout.splice(0, maxInfoLinesCount).join('')}
```

Comment:
rename to `infoLinesMaxCount`

Reply:
Fixed

Reply:
You can display the new name in the comment.